dev-tools/mage/version/version_parser.go

```diff
@@ -0,0 +1,190 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package version
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// regexp taken from https://semver.org/ (see the FAQ section/Is there a suggested regular expression (RegEx) to check a SemVer string?)
```

Comment:
Would github.com/Masterminds/semver work here? Could it be used instead of this regexp to parse the semver?

Reply:
I think it would be better to use the semver. I just brought this code from the [elastic-agent repo](https://github.com/elastic/elastic-agent/blob/main/pkg/version/version_parser.go#L16 ) just for consistency

Reply:
We can move this code from elastic-agent to elastic-agent-libs so you can just import it instead of copying it.